packages/electron/src/renderer/addBacktraceElectron.ts

```diff
@@ -0,0 +1,53 @@
+import { BacktraceCoreClientBuilder } from '@backtrace-labs/sdk-core';
+import { IpcEvents } from '../common/ipc/IpcEvents';
+import { SyncData } from '../common/models/SyncData';
+import { getIpcRpc } from './ipc/getIpcRpc';
+import { getIpcTransport } from './ipc/getIpcTransport';
+import { ConstSessionProvider } from './modules/ConstSessionProvider';
+import { IpcAsyncSessionProvider } from './modules/IpcAsyncSessionProvider';
+import { IpcBreadcrumbsStorage } from './modules/IpcBreadcrumbsStorage';
+import { IpcReportSubmission } from './modules/IpcReportSubmission';
+import { IpcRequestHandler } from './modules/IpcRequestHandler';
+import { IpcSummedMetricsQueue } from './modules/IpcSummedMetricsQueue';
+import { StubMetricsQueue } from './modules/StubMetricsQueue';
+
+export interface AddBacktraceElectronOptions {
+    readonly synchronous?: boolean;
+}
+
+export function addBacktraceElectron<T extends BacktraceCoreClientBuilder>(
+    builder: T,
+    options?: AddBacktraceElectronOptions,
+): T {
+    const ipcTransport = getIpcTransport();
+    const ipcRpc = getIpcRpc();
+
+    // Sanity check - this will throw if ping isn't listened on.
+    ipcRpc
+        .invoke(IpcEvents.ping)
+        .catch((err) =>
+            console.error(
```

Comment:
should the integration be disabled? Right now the builder will build the client in a way to forward everything to the main process, so I'm assuming it won't work and it will probably generate an infinity loop of errors?

Reply:
There's no way for me to disable the integration, and to throw this error synchronously. Least I could do is to display a big error message.

Reply:
client.disable won't work?

Reply:
This is at the building stage, the client is not built yet.